src/shared/install.c

```diff
@@ -603,6 +604,11 @@ static int remove_marked_symlinks_fd(
                         if (!found)
                                 continue;
 
+                        if (dry_run) {
+                                unit_file_changes_add(changes, n_changes, UNIT_FILE_UNLINK, p, NULL);
+                                continue;
+                        }
```

Comment:
hmm, why do you cut this short? I think the unlinkat() and rmdir_parents() calls below should simply be skipped in this case, but the mark_symlinks_for_removal() call should be called still?